content/blog/model-calibration/index.Rmd

```diff
@@ -0,0 +1,409 @@
+probabilities ranging from 60% to 70%. This indicates that the model is not creating
+predictions in this region that are consistent with the data (i.e., it is 
+under-predicting). 
+
+The number of bins in `cal_plot_breaks()` can be adjusted using `num_breaks`. 
+Here is an example of what the plot looks like if we reduce the bins from 10,
+to 5:
+
+```{r, fig.alt = "Calibration plot with 5 bins, created with the cal_plot_breaks() function"}
+segment_logistic %>% 
+  cal_plot_breaks(Class, .pred_good, num_breaks = 5 )
+```
+
+The number of breaks is a bit like a tuning parameter for these plots and should
+be based on ensuring that there is enough data in each bin to adequately estimate
+the observed event rate. If your data are small, the next version of the calibration
+plot might be a better solution. 
+
+### Windowed
+
+Another approach is to use overlapping ranges, or windows. Like the previous plot, we bin the data and calculate the event rate. However, we can add more bins by allowing them to overlap. If the data set size is small, one strategy is to use a set of wide bins that overlap one another.
+
+There are two variables that control the windows. The **step size**, controls the frequency of the windows. If we set a step size of 5%, will create a new window every 5% probability (5%, 10%, 15%... etc). The second argument is the (maximum) **window size**. If it is set to 10%, then a given step will overlap halfway into the previous step, as well as the next step. Here is a visual representation of this specific scenario:
+
+```{r, fig.alt = "Plot illustrating the horizontal location of each step and the size of the window"}
+#| echo: false
+#| fig.width: 6
+#| fig.height: 4
+#| out.width: 70%
+windows_size <- 0.10
+step_size    <- 0.05
+
+steps <- seq(0, 1, by = step_size)
+lower_cut <- steps - (windows_size / 2)
+lower_cut[lower_cut < 0] <- 0
+upper_cut <- steps + (windows_size / 2)
+upper_cut[upper_cut > 1] <- 1
+
+tibble(
+  step = steps,
+  lower = lower_cut,
+  upper = upper_cut,
+  group = seq_along(steps)
+) %>% 
+  ggplot(aes(x = step, y = group)) + 
+  geom_errorbar(aes(xmin = lower, xmax = upper)) +
+  labs(y = NULL, x = "predicted probability") +
+  theme_minimal()
+```
+
+In probably, the `cal_plot_windowed()` function provides this functionality. 
+The default step size is 0.05, and can be changed via the `step_size` argument.
+The default window size is 0.1, and can be changed via the `window_size` argument.
+
+```{r, fig.alt = "Calibration plot with 21 windows, created with the cal_plot_windowed() function"}
+segment_logistic %>% 
+  cal_plot_windowed(Class, .pred_good)
+```
+
+Here is an example of reducing the `step_size` from 0.05, to 0.02. There are
+more than double the windows:
+
+```{r, fig.alt = "Calibration plot with more steps than the default, created with the cal_plot_windowed() function"}
+segment_logistic %>% 
+  cal_plot_windowed(Class, .pred_good, step_size = 0.02)
+```
+
```

Comment:
I really appreciated how the Binned section transitioned to the Windowed one by motivating with a use case via "If your data are small, the next version...". Is there analogue here for when windowed diagnostics are helpful vs. model-based ones?

Reply:
Not sure if there's one

@topepo - Any thoughts?

Reply:
> Is there analogue here for when windowed diagnostics are helpful vs. model-based ones?

Hmm. I don't think so. I can't think of a rule that would indicate which should be used; they are just different approaches.